source/core/Variants.cpp

```diff
@@ -439,7 +439,8 @@ bool VariantsAreEqual(VariantTypeRef variantX, VariantTypeRef variantY)
 {
     DualTypeVisitor visitor;
     DualTypeEqual dualTypeEqual;
-    return visitor.Visit(variantX, variantX->Begin(kPARead), variantY, variantY->Begin(kPARead), &dualTypeEqual);
+    return (!variantX && !variantY)  // Empty variant constant with no attributes are equal
```

Comment:
Empty variant constants with no attributes are plain nullptrs. Handling that special case here. 

However, if they do go through a `SetVariantAttribute` instruction they are initialized and have attribute maps and go through the visitor.